tests/components/sleepiq/test_init.py

```diff
@@ -26,55 +24,40 @@ def mock_responses(mock, single=False):
     )
 
 
-class TestSleepIQ(unittest.TestCase):
-    """Tests the SleepIQ component."""
+async def test_setup(hass, requests_mock):
+    """Test the setup."""
+    mock_responses(requests_mock)
 
-    def setUp(self):
-        """Initialize values for this test case class."""
-        self.hass = get_test_home_assistant()
-        self.username = "foo"
-        self.password = "bar"
-        self.config = {
-            "sleepiq": {"username": self.username, "password": self.password}
-        }
-        self.addCleanup(self.tear_down_cleanup)
+    # We're mocking the load_platform discoveries or else the platforms
+    # will be setup during tear down when blocking till done, but the mocks
+    # are no longer active.
+    with patch("homeassistant.helpers.discovery.load_platform", MagicMock()):
+        assert sleepiq.setup(hass, CONFIG)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6356951)